mlguess/keras/layers.py

```diff
@@ -1,64 +1,106 @@
-import tensorflow as tf
-import numpy as np
-import tensorflow_addons as tfa
+import keras.layers as layers
+import keras.ops as ops
 
 
-#eps = np.finfo(np.float32).eps
-
-
-class DenseNormalGamma(tf.keras.layers.Layer):
-    """Implements dense layer for Deep Evidential Regression
-    
+class DenseNormalGamma(layers.Layer):
+    """
+    Implements dense output layer for a deep evidential regression model.
     Reference: https://www.mit.edu/~amini/pubs/pdf/deep-evidential-regression.pdf
     Source: https://github.com/aamini/evidential-deep-learning
+
+    Args:
+        units (int): Output size of regression tasks
+        name (str): Name of the layer
+        spectral_normalization (bool): Whether to use Spectral Normalization
+        eps (float): Minimum value for evidence (prevents potential division by zero)
+
+    Returns:
+        [mu, lambda, alpha, beta] (see reference documentation for more)
     """
     
-    def __init__(self, units, name, eps=1e-12, **kwargs):
-        super(DenseNormalGamma, self).__init__(name=name, **kwargs)
+    def __init__(self, units: int,
+                 spectral_normalization: bool = False,
+                 eps: float = 1e-12, **kwargs):
+        super().__init__(**kwargs)
         self.units = int(units)
-        self.dense = tfa.layers.SpectralNormalization(tf.keras.layers.Dense(4 * self.units, activation=None))
-        #self.dense = tf.keras.layers.Dense(4 * self.units, activation=None)
+        self.NUM_OUTPUT_PARAMS = 4
```

Comment:
If NUM_OUTPUT_PARAMS doesn't change instance to instance you may want to make it a static variable

Reply:
Thanks for adding that as a variable, much cleaner to read and understand.